### PR TITLE
rid of logex.build

### PR DESCRIPTION
**Pull request by egorka-sereda**
https://bloxxx.atlassian.net/browse/IO1-2765

### CODE REVIEW

File: logging/build.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package logging
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"log"
+	"time"
+)
+
+var parentLogger *zap.Logger
```
Comment (nkryuchkov) [inline logging/build.go:10]: I think there might be a data race if `GetLogger` is called in parallel, e.g. if we decide to run tests in parallel in the future. Consider using `atomic.Pointer`

File: logging/level.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package logging
+
+import (
+	"fmt"
+	"go.uber.org/zap/zapcore"
+)
+
+var levelEncoder zapcore.LevelEncoder
+
+// LevelEncoder takes a raw string (as []byte) and returnsthe corresponding zapcore.LevelEncoder
```
Comment (nkryuchkov) [inline logging/level.go:10]:
```suggestion
// LevelEncoder takes a raw string (as []byte) and returns the corresponding zapcore.LevelEncoder
```

File: logging/build.go (hunk 1 above)
Comment (aaron-blox) [reply logging/build.go:11]: delete this, we should use zap.L() only

File: logging/build.go [hunk 3]
```diff
@@ -2,67 +2,42 @@ package logging
...
-		EncoderConfig: zapcore.EncoderConfig{
-			MessageKey:  "message",
-			LevelKey:    "level",
-			EncodeLevel: ec.LevelEncoder,
-			TimeKey:     "time",
-			EncodeTime: func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
-				enc.AppendString(t.UTC().Format("2006-01-02T15:04:05.000000Z")) //converting time to ISO 3339 format
+var parentLogger *zap.Logger
+
+func GetLogger(level zapcore.Level) *zap.Logger {
```
Comment (aaron-blox) [inline logging/build.go:13]: make this function only set the zap.L() global logger. This function should only be called once at program invocation

Anyone that wants that logger should get it with zap.L()
```suggestion
func SetGlobalLogger(level zapcore.Level) {
```

**Review by aaron-blox — CHANGES_REQUESTED** [review]

File: protocol/v2/ssv/spectest/sync_committee_aggregator_proof_type.go [hunk 4]
```diff
@@ -17,7 +18,7 @@ import (
 func RunSyncCommitteeAggProof(t *testing.T, test *synccommitteeaggregator.SyncCommitteeAggregatorProofSpecTest) {
 	ks := testingutils.Testing4SharesSet()
 	share := testingutils.TestingShare(ks)
-	logger := logging.TestLogger(t)
+	logger := logging.GetLogger(zap.DebugLevel)
 	v := ssvtesting.BaseValidator(logger, keySetForShare(share))
```
Comment (aaron-blox) [inline protocol/v2/ssv/spectest/sync_committee_aggregator_proof_type.go:22]: so you can use like this
```suggestion
	logging.SetGlobalLogger(zap.DebugLevel)
	v := ssvtesting.BaseValidator(zap.L(), keySetForShare(share))
```

File: cli/bootnode/boot_node.go [hunk 5]
```diff
@@ -32,8 +32,9 @@ var StartBootNodeCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		loggerLevel, err := zapcore.ParseLevel(cfg.LogLevel)
-		logger := logging.Build(cmd.Parent().Short, loggerLevel, &logging.EncodingConfig{Format: cfg.GlobalConfig.LogFormat})
+		loggerLevel, err := zapcore.ParseLevel(cfg.LogLevel) //add here also
```
Comment (aaron-blox) [inline cli/bootnode/boot_node.go:35]:
```suggestion
		loggerLevel, err := zapcore.ParseLevel(cfg.LogLevel)
```

File: cli/export_keys_from_mnemonic.go [hunk 6]
```diff
@@ -19,7 +17,7 @@ var exportKeysCmd = &cobra.Command{
 	Use:   "export-keys",
 	Short: "exports private/public keys based on given mnemonic",
 	Run: func(cmd *cobra.Command, args []string) {
-		logger := logging.Build("export-keys", zap.DPanicLevel, nil)
+		logger := zap.L()
```
Comment (aaron-blox) [inline cli/export_keys_from_mnemonic.go:20]: need to set global logger here & name the logger
```suggestion
		logger := zap.L().Named("export-keys")
```

File: cli/generate_operator_keys.go [hunk 7]
```diff
@@ -1,21 +1,18 @@
...
-	"github.com/bloxapp/ssv/utils/rsaencryption"
 )
 
 // generateOperatorKeysCmd is the command to generate operator private/public keys
 var generateOperatorKeysCmd = &cobra.Command{
 	Use:   "generate-operator-keys",
 	Short: "generates ssv operator keys",
 	Run: func(cmd *cobra.Command, args []string) {
-		logger := logging.Build(RootCmd.Short, zapcore.DebugLevel, nil)
+		logger := zap.L()
```
Comment (aaron-blox) [inline cli/generate_operator_keys.go:15]: need to set global logger here & name the logger

File: cli/operator/node.go [hunk 8]
```diff
@@ -191,10 +191,8 @@ func setupGlobal(cmd *cobra.Command) (*zap.Logger, error) {
 		}
 	}
 	loggerLevel, errLogLevel := zapcore.ParseLevel(cfg.LogLevel)
-	logger := logging.Build(commons.GetBuildData(), loggerLevel, &logging.EncodingConfig{
-		Format:       cfg.GlobalConfig.LogFormat,
-		LevelEncoder: logging.LevelEncoder([]byte(cfg.LogLevelFormat)),
-	})
+	logging.SetGlobalLogger(loggerLevel)
+	logger := zap.L()
```
Comment (aaron-blox) [inline cli/operator/node.go:195]: need to set logger name

File: cli/threshold.go [hunk 9]
```diff
@@ -18,7 +16,7 @@ var createThresholdCmd = &cobra.Command{
 	Use:   "create-threshold",
 	Short: "Turns a private key into a threshold key",
 	Run: func(cmd *cobra.Command, args []string) {
-		logger := logging.Build("create-threshold", zap.DebugLevel, nil)
+		logger := zap.L()
```
Comment (aaron-blox) [inline cli/threshold.go:19]: need to set logger name

File: ekm/signer_key_manager_test.go [hunk 10]
```diff
@@ -31,7 +31,7 @@ const (
 func testKeyManager(t *testing.T) spectypes.KeyManager {
 	threshold.Init()
 
-	logger := logging.TestLogger(t)
+	logger := zap.L()
```
Comment (aaron-blox) [inline ekm/signer_key_manager_test.go:34]: we still need to use TestLogger(t), dont delete it and continue to use it

**Review by aaron-blox — CHANGES_REQUESTED** [review]

**Comment by aaron-blox** [comment]
make the above changes everywhere is relevant

**Pull request merged**